### ISSUE TITLE
Some modules cannot be marked as "browser compatible" because of outdated TS DOM types

### ISSUE THREAD

**Issue by jsejcksn** [denoland/deno_std#1973]
## Context

The current browser compatibility CI test [uses TypeScript's native DOM types](https://github.com/denoland/deno_std/blob/main/browser-compat.tsconfig.json) when type-checking modules. This is appropriate.

However, the current DOM types appear to be outdated (technically, _misaligned_ from what's being commonly shipped by browsers), so it's not possible to get tests to pass for some modules which actually _are_ compatible in that sense.

## Problem

The above described scenario prevents marking some modules as being "browser compatible" and prevents their discovery as such. I've included links to examples at the end of this message.

Refs:
- https://github.com/denoland/deno_std/pull/1972#issuecomment-1054901936
- https://github.com/denoland/deno_std/pull/1972#issuecomment-1054923976

**Comment by jsejcksn** [comment]
@crowlKats I don't see [any activity at all in the TS repo which mentions `ReadableByteStreamController`](https://github.com/microsoft/TypeScript/issues?q=ReadableByteStreamController). Am I missing something or should we file an issue?

**Comment by crowlKats** [comment]
yep, we probably should. Major missing structs are `ReadableByteStreamController` & `ReadableStreamBYOBReader`
